projects/graphql-client/src/graphql-request.service.test.ts

```diff
@@ -90,159 +154,208 @@ describe('GraphQl Request Service', () => {
-    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
-    // Make second request - should be cached
-    spectator.service
-      .queryImmediately(buildRequest(), { cacheability: GraphQlRequestCacheability.Cacheable })
-      .subscribe();
-    spectator.controller.expectNone(graphQlUri, HttpMethod.POST);
-    // Make third request but falling back to handler options now, shouldn't be cached
-    spectator.service.queryImmediately(buildRequest()).subscribe();
-    spectator.expectOne(graphQlUri, HttpMethod.POST);
-  });
+    spectator.service.query(buildRequestOne()).subscribe();
+    tick(debounceTimeMs);
+
+    expect(queryFnMock).toHaveBeenNthCalledWith(2, {
+      query: gql(requestString),
+      errorPolicy: 'all',
+      fetchPolicy: 'no-cache'
+    });
+  }));
 
   test('resubscribing to a debounced query after execution triggers a debounced new request', fakeAsync(() => {
-    spectator.service.registerHandler(buildQueryHandler({ cacheability: GraphQlRequestCacheability.NotCacheable }));
-    const query$ = spectator.service.queryDebounced(buildRequest());
+    spectator.service.registerHandler(buildQueryHandlerOne({ cacheability: GraphQlRequestCacheability.NotCacheable }));
+    const queryFnMock = mockQueryMethod();
+    const query$ = spectator.service.query(buildRequestOne());
+    const requestString = new GraphQlRequestBuilder()
+      .withSelects({ path: 'testone', children: [{ path: 'id' }, { path: 'value' }] })
+      .build();
     query$.subscribe();
-    spectator.controller.expectNone(graphQlUri, HttpMethod.POST);
+    expect(queryFnMock).not.toHaveBeenCalled();
+
     tick(debounceTimeMs);
-    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
+
+    expect(queryFnMock).toHaveBeenNthCalledWith(1, {
+      query: gql(requestString),
+      errorPolicy: 'all',
+      fetchPolicy: 'no-cache'
+    });
     flushMicrotasks();
+
     // Now two more, should trigger one more request
     query$.subscribe();
     query$.subscribe();
-    spectator.controller.expectNone(graphQlUri, HttpMethod.POST);
+    expect(queryFnMock).toHaveBeenCalledTimes(1);
     tick(debounceTimeMs);
-    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
+    expect(queryFnMock).toHaveBeenNthCalledWith(2, {
+      query: gql(requestString),
+      errorPolicy: 'all',
+      fetchPolicy: 'no-cache'
+    });
     flushMicrotasks();
     // Now one last one, one more request
     query$.subscribe();
-    spectator.controller.expectNone(graphQlUri, HttpMethod.POST);
+    expect(queryFnMock).toHaveBeenCalledTimes(2);
+
     tick(debounceTimeMs);
-    spectator.expectOne(graphQlUri, HttpMethod.POST);
+    expect(queryFnMock).toHaveBeenNthCalledWith(3, {
+      query: gql(requestString),
+      errorPolicy: 'all',
+      fetchPolicy: 'no-cache'
+    });
   }));
+
+  // test('resubscribing to a mutation triggers an immediate new request', fakeAsync(() => {
```

Comment:
This test fails for mutation. I am suspecting some changes in the way apollo creates response observable. For query it still works probably because of the `defer`. Do we need this behavior?

Reply:
the general family of resubscribe tests are to ensure the observable characteristics (in this case, being a cold unicast observable) of our service match those of a "standard" observable like those returned by httpclient. If we're mocking out apollo, since that's where the observable originates, I agree this test doesn't make a whole lot of sense (and you're correct in the wrapping `defer` being the reason this still works in the debounced query case, as we need to change the underlying characteristics to share the results there).

So we do want this behavior, but we don't need to test it here - that belongs in the int test.

Reply:
Okay.. I noticed it failed in the int test as well. I think apollo is not triggering it on resubscribe. Do you think I am missing something in the test?

Reply:
See comment on the int test - believe the issue is a missing response flush.

Reply:
So my test was wrong. Resubscribing wouldn't trigger mutate call again. The underlying observable returned from apollo would call http. But I can't test that here. Removing the test now

Reply:
agreed - the test in integration test should be sufficient.